### ISSUE TITLE
Compiler window does nothing on project

### ISSUE THREAD

**Issue by samrensenhouse** [klazuka/intellij-elm#367]
Unfortunately, I don't have a lot of extra information for this. I didn't see any info messages related to this. Let me know if there are some logs I can provide or something like that to help identify the issue.

If I click on an elm file in the project, and click the compile button no errors show up. This happens even with just syntax errors. The rest of the plugin still seems to work perfectly fine and the plugin's type system understands that it is an error.

I have used the new compiler on other projects and it works fine. One difference of note is that this project is much larger ~30k lines of code compared to the other ones.

**Comment by klazuka** [comment]
@samrensenhouse I'm assuming that this is a 0.19 application project? When you click compile, does it show "0 errors" directly next to the words "Elm Compiler"? Or is that part blank? Are there any IntelliJ notifications being shown? Or maybe you've turned off notifications for the Elm plugin?

The way this feature works is that it uses the current file in the editor to determine the Elm project that contains it. It then searches for a function named `main` whose type is either `Platform.Program` or `Html.Html`. It should correctly find `main` even if it does not have a type annotation, but you can give it a helping hand by adding an annotation if it doesn't already have one.

**Comment by samrensenhouse** [comment]
Yep this is a 0.19 project.  When I first start up the application it is blank there, when I click the build button it takes a second then shows 0 errors.

I have nothing in my intellij event log. Under appearance and behavior I have this checked if that is what you are referring to.
![image](https://user-images.githubusercontent.com/22641341/54723662-40439980-4b36-11e9-9a97-cd2585d43835.png)


This is the current main type annotation
```
main : Program () AppState Msg
main =
    Browser.element
        { view = view
        , update = Initialize.update
        , init = \_ -> init
        , subscriptions = subscriptions
        }
```

How does it search for the main function? Do I need a certain directory structure?

**Comment by samrensenhouse** [comment]
In case this is helpful info, my directory structure is like so.
```
client
│   elm.json    
└───elm
     │   Main.elm

```

I also have multiple directories in my source directories in my elm.json, could that affect this?

**Comment by klazuka** [comment]
Yes, it could be a source directory problem, especially if they are nested. I never got around to handling a case where a source directory contains (indirectly or directly) another source directory. In such cases, it may fail to identify the Elm project for the current file.

If it’s not too much trouble, can you try making the source directories siblings and see if that fixes it?

**Comment by samrensenhouse** [comment]
Figured it out. I had forgotten that there was another elm project nested inside this one that has its own Main.elm. If I comment out its main function, it works as expected.

I am probably going to try and just pull out that out into it's own separate directory later and I am assuming that will fix it for good.

**Comment by klazuka** [comment]
This is a design flaw. I created an issue to fix it (#369).
